### PR TITLE
Update client library address to github

### PR DESCRIPTION
**Pull request by liangfu**
Update client library address to github, which is more up-to-date.

### CODE REVIEW

**Comment by gunnarx** [comment]
The build failure is because the automated pipeline tries to build "nostromo-image", but master seems to still be on Miranda, so it only knows about miranda-image.  We have talked about his before though - if master is kept in sync with 12.0 we can have successful automated builds.

**Comment by tolkien-joh** [comment]
please, change the target branch from master to 12.0

File: meta-ivi/recipes-extended/persistence-client-library/persistence-client-library_1.1.0.bb [hunk 1]
```diff
@@ -9,11 +9,9 @@ LIC_FILES_CHKSUM = "file://COPYING;md5=6161c6840f21a000e9b52af81d2ca823"
 
 DEPENDS = "dlt-daemon dbus libcheck persistence-common-object"
 
-SRCREV = "fe4b73dcc282932ae3ebb8805e0b617a0016dc9a"
+SRCREV = "03fa5d83d582cb54acb4dc00a11715523d936fbf"
 SRC_URI = " \
-    git://git.projects.genivi.org/persistence/persistence-client-library.git;protocol=http \
-    file://0001-load-correct-version-of-libpers_common.patch \
-    file://0001-fix-exec-path.patch \
```
Comment (phongt) [inline meta-ivi/recipes-extended/persistence-client-library/persistence-client-library_1.1.0.bb:16]: @liangfu  Are those patches integrated to persistence-client-library repo? If yes, I think they should be cleaned up.

**Comment by tolkien-joh** [comment]
@liangfu 

even if there is a conflict for applying your patch to 12.0 branch,
 I managed to merge it my local repo.
but, the results of test script for PCL are different.
therefore, could you explain what PCL improved than before?

I tested your patch in qemux86-64 machine, and qemux86 machine contains old PCL.

[pcl.txt](https://github.com/GENIVI/meta-ivi/files/913769/pcl.txt)
[pcl.new.1.log.txt](https://github.com/GENIVI/meta-ivi/files/913780/pcl.new.1.log.txt)
[pcl.new.2.log.txt](https://github.com/GENIVI/meta-ivi/files/913781/pcl.new.2.log.txt)
[pcl.new.3.log.txt](https://github.com/GENIVI/meta-ivi/files/913784/pcl.new.3.log.txt)

[pcl.old.1.log.txt](https://github.com/GENIVI/meta-ivi/files/913785/pcl.old.1.log.txt)
[pcl.old.2.log.txt](https://github.com/GENIVI/meta-ivi/files/913786/pcl.old.2.log.txt)
[pcl.old.3.log.txt](https://github.com/GENIVI/meta-ivi/files/913787/pcl.old.3.log.txt)

**Comment by ihuerner** [comment]
Looking at the log files, persistence seems not to be working correctly at all.
In order to run the tests successfully you need to have the following preconditions fulfilled:
- A partition for persistence data must exist (size about 20 MB)
-- this partition must be mounted to /Data/mnt-c AND to /Data/mnt-wt
- A partition for backup data must exist (size about 20MB)
-- this partition needs to be mounted to /Data/mnt-backup
- Test data needs to be installed using persistence administration service (PAS)
-- Test data is available under "persistence-client-library/test/data/PAS_data.tar.gz"
-- GENIVI NSM must be started and also PAS must be started
-- install data:  "persadmin_tool install /fullPathToFile/PAS_data.tar.gz"
-- Now test data should be available under /Data/mnt-c/ and /Data/mnt-wt
- Also make sure PCL config files (e.g. pclCustomLibConfigFileTest.cfg) are available under /etc/

Now tests must run without errors.

**Comment by tolkien-joh** [comment]
@ihuerner 

there are two instance of NSM.
one is for nsm-dummy for testing
other is ordinary one.

what NSM you consider?

**Comment by ihuerner** [comment]
@tolkien-joh 

The ordinary one.

**Comment by tolkien-joh** [comment]
@ihuerner 

with latest PCO, PCL,
 I tested PCL test.

some tests failed at this time.
root@qemux86-64:/opt/tests# ./test-fw.sh persistence-client-library  
##########
persistence-client-library
##########
[ number of tests: 3 ]
[ run_1 ] 5 PASSED, 16 FAILED
[ run_2 ][ FAILED ] 11 tests.
[ run_3 ][ PASSED ]

test script is [persistence-client-library_t.inc.txt](https://github.com/GENIVI/meta-ivi/files/918759/persistence-client-library_t.inc.txt)
run_1 is the result of persistence_client_library_test
[pcl.1.0413.log.txt](https://github.com/GENIVI/meta-ivi/files/918761/pcl.1.0413.log.txt)
run_2 is the result of persistence_client_library_test_file
[pcl.2.0413.log.txt](https://github.com/GENIVI/meta-ivi/files/918764/pcl.2.0413.log.txt)

one this I think it's weired is the contents of /Data/mnt-c
root@qemux86-64:/opt/tests# ls /Data/mnt-c/
browser(null)1(null)1
lt-persistence_client_library_test(null)1(null)1
lt-persistence_client_library_test(null)200(null)100
lt-persistence_client_library_test(null)4(null)12
lt-persistence_client_library_test(null)99(null)99

should it be like below?
browser
lt-persistence_client_library_test

**Comment by ihuerner** [comment]
@tolkien-joh 

Not it should not look like this, still something wrong in setting up test data using PAS.
What is the result when running just "persadmin_tool install /Data/mnt-backup/PAS_data.tar.gz".
Please make sure that NSM and PAS is running before calling persadmin_tool install.

By the way, you can skip test persistence_client_library_dbus_test

**Comment by tolkien-joh** [comment]
@ihuerner 

I did following after boot:
systemctl restart pas-daemon
systemctl restart nodestatemanager-daemon
persadmin_tool install /Data/mnt-backup/PAS_data.tar.gz

then ls -lR...
[ls-lR.txt.gz](https://github.com/GENIVI/meta-ivi/files/918889/ls-lR.txt.gz)

**Comment by ihuerner** [comment]
@tolkien-joh 

This looks good, test data seems to be installed correctly,
If you now run persistence tests, like "./persistence_client_library_test",
what is the result/console output?

**Comment by tolkien-joh** [comment]
@ihuerner 

here is...
root@qemux86-64:/opt/tests/persistence-client-library# ./persistence_client_library_test
INFO: Skipping testcase MultiThreadedRead  (0x402dea)
INFO: Skipping testcase MultiThreadedWrite (0x402c31)
Running suite(s): Persistence Client Library (Key-API)
23%: Checks: 21, Failures: 16, Errors: 0
persistence_client_library_test.c:1547:P:NoPluginFunc:test_NoPluginFunc:0: Passed
persistence_client_library_test.c:285:F:SetData:test_SetData:0: Wrong write size
persistence_client_library_test.c:139:F:GetData:test_GetData:0: Assertion '(char*)buffer == "CACHE_ +48 10' 38.95, +8 44' 39.06"' failed: (char*)buffer == "", "CACHE_ +48 10' 38.95, +8 44' 39.06" == "CACHE_ +48 10' 38.95, +8 44' 39.06"
persistence_client_library_test.c:218:F:GetDataHandle:test_GetDataHandle:0: Buffer not correctly read => 1
persistence_client_library_test.c:421:F:SetDataNoPRCT:test_SetDataNoPRCT:0: Wrong write size
persistence_client_library_test.c:450:F:GetDataSize:test_GetDataSize:0: Invalid size
persistence_client_library_test.c:480:F:DeleteData:test_DeleteData:0: Failed to delete key
persistence_client_library_test.c:591:P:DataHandleOpen:test_DataHandleOpen:0: Passed
persistence_client_library_test.c:672:F:ReadDefault:test_ReadDefault:0: Assertion 'ret == strlen("DEFAULT_01!")' failed
persistence_client_library_test.c:702:F:ReadConfDefault:test_ReadConfDefault:0: Assertion 'ret == strlen("CONF_DEFAULT_01!")' failed
persistence_client_library_test.c:730:F:WriteConfDefault:test_WriteConfDefault:0: Write Conf default data: write size does not match
persistence_client_library_test.c:852:P:NegHandle:test_NegHandle:0: Passed
persistence_client_library_test.c:871:F:UTF-8:test_utf8_string:0: Wrong read size
persistence_client_library_test.c:888:F:Notifications:test_Notifications:0: Failed to register
persistence_client_library_test.c:606:F:Plugin:test_Plugin:0: Assertion 'ret == strlen("Custom plugin -> plugin_get_data: secure!")' failed
persistence_client_library_test.c:1041:F:SharedAccess:test_SharedAccess:0: Failed to write shared data 
persistence_client_library_test.c:1084:F:VO722:test_VO722:0: Wrong write size
persistence_client_library_test.c:1182:P:InvalidPluginfConf:test_InvalidPluginfConf:0: Passed
persistence_client_library_test.c:1174:P:NoRct:test_NoRct:0: Passed
persistence_client_library_test.c:800:F:InitDeinit:test_InitDeinit:0: Lifecycle set NOT allowed, but should
persistence_client_library_test.c:1215:F:SharedData:test_SharedData:0: Wrong write size

**Comment by ihuerner** [comment]
@tolkien-joh 

Are there any DLT logs available?
How does the content from the file "/etc/pclCustomLibConfigFileTest.cfg" looks like?
Is the test running with root rights?

**Comment by tolkien-joh** [comment]
@ihuerner 

the contenct of /etc/pclCustomLibConfigFileTest.cfg are...
hwinfo /usr/lib/libhwinfoperscustom.so init async
secure /usr/lib/libsecureperscustom.so od sync
custom3 /usr/lib/libcustom3perscustom.so init sync
default /usr/lib/libpers_common.so init sync
emergency /usr/lib/libemergencyperscustom.so od async
early /usr/lib/libearlyperscustom.so od sync
anInvalidEntry /wrongPath/wrongLib do cnys
custom2 /usr/lib/libcustom2perscustom.so init sync

and, I'll reboot system for getting dlt.log.

**Comment by tolkien-joh** [comment]
@ihuerner 

yes. I ran this test with root account.
there is no something on dlt.log

[  248.556879] DLT   386 DEBUG     ContextID 'PCL' registered for ApID 'PCLT', Description=Ctx for PCL Logging
[  248.557345] DLT   386 NOTICE    Send log-level to context: PCLT:PCL [-1 -> 4] [-1 -> 0]
[  248.675670] DLT   386 DEBUG     ApplicationID 'NSM' registered for PID 315, Description=Node State Manager
[  248.676357] DLT   386 DEBUG     ContextID 'Pers' registered for ApID 'NSM', Description=PersCommonLLDB
[  248.676490] DLT   386 NOTICE    Send log-level to context: NSM:Pers [-1 -> 4] [-1 -> 0]
[  248.676955] DLT   386 DEBUG     ContextID '005' registered for ApID 'NSM', Description=Context for the NSM
[  248.680988] DLT   386 NOTICE    Send log-level to context: NSM:005 [-1 -> 4] [-1 -> 0]
[  248.681863] DLT   386 DEBUG     ContextID 'PCL' registered for ApID 'NSM', Description=Ctx for PCL Logging
[  248.681963] DLT   386 NOTICE    Send log-level to context: NSM:PCL [-1 -> 4] [-1 -> 0]
[  248.697671] DLT   386 DEBUG     ApplicationID 'PERS' registered for PID 301, Description=Persistence Administration Service
[  248.697822] DLT   386 DEBUG     ContextID 'PADM' registered for ApID 'PERS', Description=Persistence Administration Service Context
[  248.697930] DLT   386 NOTICE    Send log-level to context: PERS:PADM [-1 -> 4] [-1 -> 0]
[  248.707630] DLT   386 DEBUG     ContextID 'PCOM' registered for ApID 'PERS', Description=PersistenceCommonIPC Context
[  248.707747] DLT   386 NOTICE    Send log-level to context: PERS:PCOM [-1 -> 4] [-1 -> 0]
[  248.717664] DLT   386 DEBUG     Unregistered CtID 'PCL' for ApID 'PCLT'
[  248.734351] DLT   386 DEBUG     ContextID 'PCL' registered for ApID 'PCLT', Description=Ctx for PCL Logging
[  248.734472] DLT   386 NOTICE    Send log-level to context: PCLT:PCL [-1 -> 4] [-1 -> 0]
[  248.837385] DLT   386 DEBUG     Unregistered CtID 'PCL' for ApID 'PCLT'
[  248.844761] DLT   386 DEBUG     ContextID 'PCL' registered for ApID 'PCLT', Description=Ctx for PCL Logging
[  248.844876] DLT   386 NOTICE    Send log-level to context: PCLT:PCL [-1 -> 4] [-1 -> 0]
[  248.942420] DLT   386 DEBUG     Unregistered CtID 'PCL' for ApID 'PCLT'
[  248.944780] DLT   386 DEBUG     ContextID 'PCL' registered for ApID 'PCLT', Description=Ctx for PCL Logging
[  248.944888] DLT   386 NOTICE    Send log-level to context: PCLT:PCL [-1 -> 4] [-1 -> 0]

**Comment by ihuerner** [comment]
@tolkien-joh 

Is the all DLT output you get when running persistence tests?

**Comment by tolkien-joh** [comment]
@ihuerner 

I changed log output from stdout to file (/tmp/dlt.log) in /etc/dlt.conf .
that's all of dlt.log file.

**Comment by ihuerner** [comment]
@tolkien-joh 

There must be more available in the dlt, is is possible to get the output form dlt-viewer?

**Comment by tolkien-joh** [comment]
@ihuerner 

since there was no dlt-log receiver, there was log message from application.
after running "systemctl start dlt-receive". I've got a lot of logs.

[dlt-receive.log.txt.gz](https://github.com/GENIVI/meta-ivi/files/919609/dlt-receive.log.txt.gz)

**Comment by ihuerner** [comment]
@tolkien-joh 

Here is the root cause: 
3169 2017/04/13 12:45:02.967007    8349813 017 ECU1 PCLT PCL- log error V 2 [load_custom_plugins => E r r o r could not load plugin:  /usr/lib/libpers_common.so]

The default database backend can't be loaded, according to the config file /etc/pclCustomLibConfigFileTest.cfg the default plugin is expected to be under the following location:
/usr/lib/libpers_common.so

Could you please make sure the library is available under this location, or adjust the config file with the correct path.

**Comment by tolkien-joh** [comment]
@ihuerner 

there is no /usr/lib/libpers_common.so at /usr/lib/
root@qemux86-64:/opt/tests/persistence-client-library# ls /usr/lib/libpers*
/usr/lib/libpers_admin_access_lib.so.1
/usr/lib/libpers_admin_access_lib.so.1.0.0
/usr/lib/libpers_common.so.0
/usr/lib/libpers_common.so.0.1.1
/usr/lib/libpersistence_client_library.so.7
/usr/lib/libpersistence_client_library.so.7.1.1

therefore, I change the name of plugin to 
default /usr/lib/libpers_common.so.0 init sync

then, 3 tests are passed.
[pcl.1.0413.2.log.txt](https://github.com/GENIVI/meta-ivi/files/919712/pcl.1.0413.2.log.txt)

**Comment by ihuerner** [comment]
@tolkien-joh 

Could you please provide me again the DLT logs, these are very helpful to figure out what's wrong.

**Comment by tolkien-joh** [comment]
@ihuerner 

here it is.
[dlt-receive.2.log.txt.gz](https://github.com/GENIVI/meta-ivi/files/919736/dlt-receive.2.log.txt.gz)

current test environment is qemux86-64, and
the recipe I used is
https://github.com/GENIVI/meta-ivi/blob/12.0/meta-ivi/recipes-extended/persistence-client-library/persistence-client-library_1.1.0.bb

**Comment by ihuerner** [comment]
@tolkien-joh 

Something is wrong with the following database: 
[KISSDB_open : Opening database file: < /Data/mnt-c/lt-persistence_client_library_test/cached.itz > failed:  No such file or directory]

Looking at the previous provided ls command, /Data/mnt-c is empty:
/Data/mnt-c:
total 0

You need to make sure to mount the persistence partition to /Data/mnt-wt AND /Data/mnt-c!!
Persistence requires one data partition mounted to two different mountpoints.

**Comment by tolkien** [comment]
@ihuerner 

I found what is wrong.
I simply created /Data/tmp directory, and
mount it to mnt-c and mnt-wt with bind option.
then, It is passed!

root@qemux86-64:/opt/tests/persistence-client-library# ./persistence_client_library_test
INFO: Skipping testcase MultiThreadedRead  (0x402dea)
INFO: Skipping testcase MultiThreadedWrite (0x402c31)
Running suite(s): Persistence Client Library (Key-API)
100%: Checks: 21, Failures: 0, Errors: 0

I have to modify test script.

**Comment by tolkien-joh** [comment]
@ihuerner 

upload the patch
see https://github.com/GENIVI/meta-ivi/commit/959dfa43862d496c26600fe7dacb4139fc7a592d

**Comment by ihuerner** [comment]
@tolkien 

Great that it is working now!

**Pull request closed without merging**